feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts

```diff
@@ -330,48 +332,88 @@ export class ConfiguratorGroupMenuComponent {
     } else if (this.isForwardsNavigation(event)) {
       if (group && this.hasSubGroups(group)) {
         this.click(group);
+        this.setFocusForSubGroup(group, currentGroup?.id);
       }
     } else if (this.isBackNavigation(event)) {
       if (this.configGroupMenuService.isBackBtnFocused(this.groups)) {
         this.navigateUp();
+        this.setFocusForMainMenu(currentGroup?.id);
       }
     }
   }
 
+  /**
+   * Persists the keyboard focus state for the given key
+   * from the main group menu by back navigation.
+   *
+   * @param {string | undefined} currentGroupId - Current group ID
+   */
+  setFocusForMainMenu(currentGroupId: string | undefined): void {
+    let key: string | undefined = currentGroupId;
+    this.configuration$.pipe(take(1)).subscribe((configuration) => {
+      configuration?.groups?.forEach((group) => {
+        if (
+          group?.subGroups?.length !== 1 &&
+          (this.isGroupSelected(group?.id, currentGroupId) ||
+            this.containsSelectedGroup(group, currentGroupId))
+        ) {
+          key = group?.id;
+        }
+      });
+    });
+    this.configUtils.setFocus(key);
+  }
+
+  /**
+   * Persists the keyboard focus state for the given key
+   * from the subgroup menu by forwards navigation.
+   *
+   * @param {Configurator.Group} group - Group
+   * @param {string | undefined} currentGroupId - Current group ID
+   */
+  setFocusForSubGroup(
+    group: Configurator.Group,
+    currentGroupId: string | undefined
```

Comment:
Also shorten to 
```suggestion
    currentGroupId?: string 
```
please

Reply:
Done!